overlord/snapstate/autorefresh.go

```diff
@@ -768,21 +762,22 @@ func inhibitRefresh(st *state.State, snapst *SnapState, snapsup *SnapSetup, info
 	// Decide on what to do depending on the state of the snap and the remaining
 	// inhibition time.
 	now := time.Now()
+	// cannot inhibit refreshes for more than maxInhibition;
 	switch {
 	case snapst.RefreshInhibitedTime == nil:
 		// If the snap did not have inhibited refresh yet then commence a new
 		// window, during which refreshes are postponed, by storing the current
 		// time in the snap state's RefreshInhibitedTime field. This field is
 		// reset to nil on successful refresh.
 		snapst.RefreshInhibitedTime = &now
-		busyErr.timeRemaining = (maxInhibition - now.Sub(*snapst.RefreshInhibitedTime)).Truncate(time.Second)
+		busyErr.timeRemaining = (MaxInhibitionTime(st) - now.Sub(*snapst.RefreshInhibitedTime)).Truncate(time.Second)
```

Comment:
Perhaps get `MaxInhibitionTime` somewhere aside as it has a bunch of transaction/parsing logic and seems to be used in several places here.